src/core.c/CompUnit/Repository/FileSystem.pm6

```diff
@@ -289,7 +289,9 @@ class CompUnit::Repository::FileSystem
     # but probably a poorly formed one.
     method !dist-from-ls {
         my $prefix := self!files-prefix;
-        my &ls := { Rakudo::Internals.DIR-RECURSE($_).map(*.IO) }
+        my $SPEC := $*SPEC;
+        my $CWD := $*CWD;
+        my &ls := { Rakudo::Internals.DIR-RECURSE($_).map({ IO::Path.new($_, :$SPEC, :$CWD) }) }
         my &to-relative := { $_.relative($prefix).subst(:g, '\\', '/') }
```

Comment:
It occurs to me that we could probably pass in `:$CWD` to `relative` here. 

Also, we could also probably apply a similar `IO::Path.new(...)` replacement to the `'.' ~ .IO.extension` that occurs a little later on (although maybe we should just parse the extension off the string without converting it to an `IO::Path`).

Reply:
Yeah, I thought there would be more places to do this, but hadn’t yet looked exactly where.

Reply:
Ah I was wrong - the positional argument `$prefix` is already setting `$CWD`